mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1318,7 +1314,7 @@ struct GridwiseGemmV2RewritePattern
     auto loc = op.getLoc();
 
     // Obtain data type.
-    auto elementType = op.c().getType().cast<MemRefType>().getElementType();
+    auto elementType = op.b().getType().cast<MemRefType>().getElementType();
```

Comment:
I happen to need to change this as well because the kernel return arg C type is different with A or B matrix. However, how does the fp16 not using atomic add being able to work before (I'm assuming C type is fp32, so the old code sort of give us the element type of fp32?)

Reply:
on the non-atomic add version, because each thread stores to a different location on the global memref, we do an in-place fp32->fp16 truncation before storing it.